Tasks/XcodeV5/xcode.ts

```diff
@@ -115,28 +118,34 @@ async function run() {
         let packageApp: boolean = tl.getBoolInput('packageApp', true);
         let args: string = tl.getInput('args', false);
 
+        telemetryData.actions = actions;
+        telemetryData.packageApp = packageApp;
+
         //--------------------------------------------------------
         // Exec Tools
         //--------------------------------------------------------
 
         // --- Xcode Version ---
         let xcv: ToolRunner = tl.tool(tool);
         xcv.arg('-version');
-        let xcodeVersion: number = 0;
+        let xcodeMajorVersion: number = 0;
         xcv.on('stdout', (data) => {
             let match = data.toString().trim().match(/Xcode (.+)/g);
             tl.debug('match = ' + match);
             if (match) {
-                let version: number = parseInt(match.toString().replace('Xcode', '').trim());
-                tl.debug('version = ' + version);
-                if (!isNaN(version)) {
-                    xcodeVersion = version;
+                const versionString = match.toString().replace('Xcode', '').trim();
+                let majorVersion: number = parseInt(versionString);
```

Comment:
Is it worth doing the trim and parse here? Why not handle these when analyzing the telemetry?

Reply:
+1, I see you are logging the versionString which is the full version. I like putting full version in the telemetry. Is there a reason to get the major version here - was there an issue or is it an improvement?

Reply:
`majorVersion` and `xcodeMajorVersion` are just variable renames. When I added version string I wanted to make a distinction between which variables had the full version number and which had the major number.

The telemetry property looks like this: `"xcodeVersion":"8.3.3"` (no "Xcode " prefix)

The major version number is used to handle auto export and iCloud below.